### ISSUE TITLE
Passed decimal parameter (i.e : double, float) converted the separator into comma (,) instead of dot (.)

### ISSUE THREAD

**Issue by ElementalCyclone** [sqlkata/querybuilder#205]
I think this have something to do with locale or CultureInfo formatting

`var que = new Query("UserTable").Where("ID", usrID).AsUpdate(setParam);` where `setParam` is an `Dictionary<string, object>` with this content

    {"Val1", -7.79607},
    {"Val2", 110.362633 }

printing out the Query from `QueryCompiler.Compile(que).ToString();` , where `QueryCompiler` is an `SqlKata.Compiler.SqlServerCompiler` object, into the Debug output window, outputting this :

    UPDATE [UserTable] SET [Val1] = -7,79607, [Val2] = 110,362633 WHERE [ID] = -23323

Which is invalid for SQL string, as comma is used for separating parameters not decimal in number. It's supposed to be look like this : 

    UPDATE [UserTable] SET [Val1] = -7.79607, [Val2] = 110.362633 WHERE [ID] = -23323

printing out each of `QueryCompiler.Compile(que).NamedBinding` and `QueryCompiler.Compile(que).Bindings` also resulted same

As per this issue posted, I've checked the documentation here [https://sqlkata.com/docs/compilers](https://sqlkata.com/docs/compilers) but there's no hint, clue, nor topics about working with floating numbers.

This issue might heavily impacts developers who opt-out to not choose using SqlKata.Execution and prefer to just get the command string from `ToString()` yet have to works with floating numbers.

------------
### Suggested Solution
Allow `ToString()` method from SqlKata's compilers to accept `IFormatProvider` overload, thus, when the conversion process hitting the number object, it can be correctly converted with `CultureInfo.InvariantCulture`, regardless the server or development machine current `CultureInfo`

**Comment by ahmad-moussawi** [comment]
Yes, I think we should force the formatting output to be the same on all culture/environments, so no need to pass the CultureInfo from the client (this needs more investigations)

Secondly, You should not use `ToString()` to execute your queries at all, the only purpose of ToString() is to make the debugging easier.

**Comment by ElementalCyclone** [comment]
Yes, as you've stated, we are not supposed to use `ToString()` to output the command string other than for debugging purpose.

But those who opt to not use `.Execution` package then have no ways to pull out the actual complete command string

**Comment by ceastwood** [comment]
@ElementalCyclone 

The complete command string can be retrieved by the .Sql property of SqlResult. I put together a quick set of code to help (hopefully it does).

```csharp

Thread.CurrentThread.CurrentCulture = CultureInfo.GetCultureInfo("id");

var compiler = Compilers.Get<SqlServerCompiler>(EngineCodes.SqlServer);

var setParam = new ReadOnlyDictionary<string, object>(new Dictionary<string, object>()
{
    ["Val1"] = -7.79607m,
    ["Val2"] = 110.36233m
});

int userTableID = 1;


var query = new Query("UserTable")
    .Where("ID", userTableID)
    .AsUpdate(setParam);

var result = compiler.Compile(query);

//CurrentCulture dictates usage of comma
Assert.Contains("-7,79607", result.ToString());

//Sql statement
Assert.Equal("UPDATE [UserTable] SET [Val1] = @p0, [Val2] = @p1 WHERE [ID] = @p2", result.Sql);

//Bindings
Assert.Equal(setParam["Val1"], result.Bindings[0]);
Assert.Equal(setParam["Val2"], result.Bindings[1]);
Assert.Equal(userTableID, result.Bindings[2]);


//With en-us culture
Thread.CurrentThread.CurrentCulture = CultureInfo.GetCultureInfo("en-us");
result = compiler.Compile(query);

Assert.Contains("-7.79607", result.ToString());

```

**Comment by ElementalCyclone** [comment]
@ceastwood, thank you for your response.

In my case, unfortunately, i can't (to be exact, heavily discouraged by the environment) to do changes to `Thread.CurrentThread`, so the current working workaround for me is to make another function that replace every each '@p1', '@p2', '@p3' and so on to the correct string representation

```c#
    internal static string CompiledString(Query que)
    {
        var compiled = QueryCompiler.Compile(que);
        if(compiled.Bindings.Count <= 0)
        {
            // Return as-is, as the SQL query doesn't have any parameter
            // that need to be formatted
            return compiled.ToString();    
        }
        else
        {
            // Iterate through whole Bindings, replacing the respective "@p" placeholders
            string returnPlaceholder = compiled.Sql;
            for(int i = 0; i < compiled.Bindings.Count; i++)
            {
                var para = compiled.Bindings[i];
                var replacer = "";
                if (para is string || para is char)
                {
                    replacer = $"'{para.ToString()}'";
                }
                else if (para is double)
                {
                    //in short, this is the line that solve my prblem
                    replacer = ((double)para).ToString(CultureInfo.InvariantCulture);
                }
                else
                {
                    replacer = para.ToString();
                }
                returnPlaceholder = returnPlaceholder.Replace("@p" + 1, replacer);
            }
            return returnPlaceholder;
        }
    }
```
Also, i think i'll just close this issue since workarounds are exist now.

Thank you.

**Comment by ahmad-moussawi** [comment]
@ElementalCyclone, I think you provided the solution for the issue, would you mind to do a PR so we can merge it with the code base.
